From 863347d2ea86fa3e749a6c6a1fabcbc943ba616b Mon Sep 17 00:00:00 2001
From: Don Darling <don.osc2@gmail.com>
Date: Mon, 12 Apr 2010 13:29:41 -0500
Subject: [PATCH 04/12] Try progressive mode first for Component input sources.

The v4l2src driver typically tries to initialize a capture input in interlaced
mode first, and if that fails it tries progressive mode.  When using Component
inputs on TI devices, the driver expects that we get this setting right on the
first try.

This change tries progressive mode first when Component input was specified
explictly via the input-src property.
---
 sys/v4l2/gstv4l2object.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 50d7204..48af22c 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -1863,10 +1863,15 @@ gst_v4l2_object_set_format (GstV4l2Object * v4l2object, guint32 pixelformat,
    * correctly the first time we call S_FMT, based on the capture device we are
    * using.  Override settings here where appropriate.
    */
-  #if defined(Platform_dm6467) || defined(Platform_dm365)
+  #if defined(Platform_dm6467)
     format.fmt.pix.field = V4L2_FIELD_NONE;
   #endif
 
+  /* Try progressive mode first for Component input sources */
+  if(v4l2object->input_src && !strcmp(v4l2object->input_src, "COMPONENT")) {
+    format.fmt.pix.field = V4L2_FIELD_NONE;
+  }
+
   if (v4l2_ioctl (fd, VIDIOC_S_FMT, &format) < 0) {
     if (errno != EINVAL)
       goto set_fmt_failed;
-- 
1.6.3.3

